pypac/api.py

```diff
@@ -162,7 +171,10 @@ def __init__(self, pac=None, proxy_auth=None, pac_enabled=True,
             :class:`requests.exceptions.ProxyError` are matched.
         :param str socks_scheme: Scheme to use when PAC file returns a SOCKS proxy. `socks5` by default.
         """
-        super(PACSession, self).__init__()
+        if session:
+            session.__class__ = PACSession
+        else:
+            super(PACSession, self).__init__()

```

Comment:
I still don't see why `session` needs to be added to `__init__()` or why `__class__` is being modified. If we don't change `__init__()` at all, isn't it still possible to reach your goal? Example:

```
s = Session()
s.headers['User-Agent'] = 'foobar'
p = PACSession()
p.get_pac(session=s)
```

Aside from this, your other changes make sense to me.

Reply:
Yes, it not exactly about setting headers.
Session is needed if we want to "upgrade" (cast) our existing `requests.Session` (or other Session) to the PACSession.
Without this we should set `session.proxies` "manually" before every `session.put` and `session.get`. Of course it's make sense only if we have tricky pac file as https://antizapret.prostovpn.org/proxy.pac.

Reply:
I'm not familiar with this pattern. How important is it to your work? If it's not necessary, I would prefer to exclude it for now.

Reply:
It is not important. It can be excluded.
And you right, because “casting”, that I use, is not completely secure.

Reply:
Okay, please undo the changes to `__init__()`, and I'll merge this pull request.

Reply:
Done